Main/src/main/java/com/main/acad/dao/ChapterDao.java

```
@@ -22,7 +25,8 @@
     private String sqlUpdate = "UPDATE chapters SET name =? WHERE id_chapter=?";
     private String sqlRemove = "DELETE  FROM chapters WHERE id_chapter=?";
     private String sqlGetByid = "SELECT r.id_chapter, r.id_refrence, c.name FROM \"references\" r INNER JOIN chapters c ON r.id_refrence = c.id_chapter where c.id_chapter = ?";
-    private String sqlGetAll = "SELECT r.id_chapter, r.id_refrence, c.name FROM \"references\" r INNER JOIN chapters c ON r.id_refrence = c.id_chapter";
+    private String sqlGetAll = "SELECT * FROM \"chapters\" c INNER JOIN \"references\" r ON r.id_chapter = c.id_chapter where r.id = r.id_chapter";
+    private String sqlGetChild = "select name from chapters where id_chapter IN (select r.id from chapters c inner join \"references\" r ON r.id_chapter = c.id_chapter where r.id_chapter = ?  and r.id != ?)";
 
     @Override
     public void addChapter(Chapter chapter) {
```

for methods that not implemented please throw new UnsupportedOperationException